lib/app/modules/restaurant_home/pages/product/add_product_card.dart

```diff
@@ -1,24 +1,115 @@
-      hoverColor: Colors.transparent,
-      child: const SizedBox(
-        height: 200,
-        width: 400,
-        child: Icon(
-          Icons.add_outlined,
-          color: secondaryColor,
-          size: 40,
+class AddProduct extends StatefulWidget {
+  const AddProduct({Key? key}) : super(key: key);
+  @override
+  _AddProductState createState() => _AddProductState();
+}
+
+class _AddProductState extends State<AddProduct> {
+  final RestaurantHomeStore store = Modular.get();
+  @override
+  Widget build(BuildContext context) {
+    Size screen = MediaQuery.of(context).size;
+    return Card(
+      shape: RoundedRectangleBorder(
+        borderRadius: BorderRadius.circular(10),
+      ),
+      child: InkWell(
+        borderRadius: BorderRadius.circular(10),
+        hoverColor: Colors.transparent,
+        child: const SizedBox(
+          height: 200,
+          width: 400,
+          child: Icon(
+            Icons.add_outlined,
+            color: secondaryColor,
+            size: 40,
+          ),
         ),
+        onTap: () {
+          showDialog(
+            context: context,
+            builder: (context) {
+              return SizedBox(
+                child: AlertDialog(
+                  title: const Text(
+                    "Cadastrar novo produto",
+                    textAlign: TextAlign.center,
+                    style: TextStyle(
+                      fontFamily: 'Nunito',
+                      color: Colors.black,
+                      fontSize: 16,
+                      fontWeight: FontWeight.w300,
+                    ),
+                  ),
+                  actions: <Widget>[
+                    Center(
+                      child: Column(
+                        crossAxisAlignment: CrossAxisAlignment.center,
+                        children: [
+                          const ProductImage(),
+                          CustomTextField(
+                            controller: store.formProduct['name'],
+                            label: 'Nome do produto',
+                          ),
+                          CustomTextField(
+                            controller: store.formProduct['desc'],
+                            label: 'Descrição do produto',
+                          ),
+                          PriceTextField(
+                            controller: store.formProduct['price'],
+                            label: 'Preço do produto',
+                          ),
+                          CustomTextField(
+                            controller: store.formProduct['categories'],
+                            label: 'Categoria do produto',
+                          ),
+                        ],
+                      ),
+                    ),
+                    Center(
+                      child: CustomButton(
+                        label: "Salvar",
+                        onPressed: () async {
+                          store.cadastroProduto();
```

Comment:
O botão de salvar precisa executar a ação de cadastro de produto sob a condição de todos os campos estarem validados. Por mais que a validação apareça na tela - "esse campo não pode ficar vazio" - a ação é executada independentemente e o produto adicionado ao firebase. Quando o salvar for executado com sucesso (lançar mensagem de confirmação - pode ser um flushbar como o de quando a imagem é inválida), os valores dos text editing tem que ser limpos (caso contrário, os valores vão ficar salvo ao abrir pra adicionar outro produto). Acredito que isso seja tudo.